### PR TITLE
MDM-535 Check parameters and report errors

### PR DESCRIPTION
**Pull request by ryanjdew**
Ensure we have errors that explain the required parameters for REST extensions. 

Add documentation to the REST APIs that detail the parameters and use that for a unit test that checks that an error is thrown for a missing required parameter.

### CODE REVIEW

File: src/main/ml-modules/services/metadata/sm-match.xml [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+              <li>rs:preview (xs:boolean?)</li>
+              <li>rs:start (xs:integer?)</li>
+              <li>rs:pageLength (xs:integer?)</li>
+            </ul>
+          </dd>
+          <dt>Example Request Body (application/json)</dt>
+          <dd>
+            <code>
+              {
+              "document": {
```
[the remark this answers is not in the document]
Comment (ryanjdew) [reply src/main/ml-modules/services/metadata/sm-match.xml]: Adjusted the uri parameter to be optional and added additional test code for the corner case of requiring uri or document.
